jupyter_book/config.py

```diff
@@ -333,3 +345,30 @@ def _recursive_update(config, update, list_extend=False):
                 config[key] = val
         else:
             config[key] = val
+
+
+def _get_files_outside_toc(
+    toc: Path, sourcedir: Path, excluded_patterns: Collection[str]
+):
+    source_root = sourcedir or Path()
+    source_files = {f for f in glob(str(source_root / "**/*"), recursive=True)}
+
+    excluded_file_sets = [set(glob(p, recursive=True)) for p in excluded_patterns]
+    included_files: Set[str] = {
+        Path(relpath(f, source_root)).as_posix()
+        for f in source_files.difference(*excluded_file_sets)
+        if not isdir(f)
+    }
+
+    toc_yaml = yaml.safe_load(toc.read_text(encoding="utf8"))
+
+    from nested_lookup import nested_lookup
+
+    toc_files = {f for f in nested_lookup("file", toc_yaml)}
```

Comment:
nested lookup is cool! I guess it is probably fairly lightweight? In general we want to be conservative about adding dependencies - do you think the extra dep is worth the functionality?

(note that if we do add the dep, there are probably other places in the codebase where it'd be useful as we are doing some recursive stuff here and there I think)

Reply:
Agree on being conservative.

It only depends on one lib [reqs](https://github.com/russellballestrini/nested-lookup/blob/master/requirements.txt)

I reckon it's slightly safer to use this than reimplement the algorithm assuming we are sufficiently pinned (may want to change to `==` from `~=`, but that has risks too..)